spec/filters/truncate_spec.rb

```diff
@@ -0,0 +1,88 @@
+# encoding: utf-8
+
+require "logstash/devutils/rspec/spec_helper"
+require "logstash/filters/truncate"
+require "flores/random"
+require "flores/rspec"
+
+RSpec.configure do |config|
+  Flores::RSpec.configure(config)
+end
+
+
+describe LogStash::Filters::Truncate do
+  analyze_results
+
+  context "defaults" do
+    let(:data) {
+      {
+        "foo" => { "bar" => Flores::Random.text(0,1000) },
+        "one" => { "two" => { "three" => Flores::Random.text(0,1000) } },
+        "baz" => Flores::Random.text(0,1000),
+      }
+    }
```

Comment:
Indeed! This is useful to help make sure we truncate to byte-length and then round down to the nearest utf-8 character.